### PR TITLE
League stats refactor

### PR DESCRIPTION
**Pull request by rrabinovitch**
@evilaspaas1 @gabichuelas @sciencefixion 

I used this branch to make my helper methods more dynamic. They can now take arguments, which minimizes repetition across helper methods that were getting to be really similar.

If any of your stats method categories have similarly repetitive calculations, this could be a great way ensure our code is DRY as we dive into TheBigRefactor

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -117,143 +116,77 @@ def find_team_by_id(id)
...
-
-    lowest_scoring_visitor_id = avgs_by_team.min_by do |_visiting_team_id, avg_score|
-      avg_score
-    end.first
-
-    find_team_by_id(lowest_scoring_visitor_id).team_name
+    away_teams = find_game_teams("away")
+    sorted_away_team_scores = sort_scores_by_team(away_teams)
+    team_avgs = team_averages(sorted_away_team_scores)
+    find_team_by_id(team_with_lowest_average_score(team_avgs)).team_name
```
Comment (gabichuelas) [inline lib/stat_tracker.rb:179]: @rrabinovitch I can't tell if this is indented properly. In the review it looks like it's outside of `def lowest_scoring_visitor`

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -117,143 +116,77 @@ def find_team_by_id(id)
...
-
-    lowest_scoring_home_id = avgs_by_team.min_by do |_visiting_team_id, avg_score|
-      avg_score
-    end.first
-
-    find_team_by_id(lowest_scoring_home_id).team_name
+    home_teams = find_game_teams("home")
+    sorted_home_team_scores = sort_scores_by_team(home_teams)
+    team_avgs = team_averages(sorted_home_team_scores)
+    find_team_by_id(team_with_lowest_average_score(team_avgs)).team_name
```
Comment (gabichuelas) [inline lib/stat_tracker.rb:186]: @rrabinovitch same here

File: test/stat_tracker_test.rb [hunk 3]
```diff
@@ -169,6 +186,20 @@ def test_it_can_identify_worst_offense
...
+      assert_equal "away", game_team.hoa
+    end
+
+    @stat_tracker.find_game_teams("home").each do |game_team|
+      assert_equal "home", game_team.hoa
+    end
+
+    assert_equal 3, @stat_tracker.find_game_teams("away").count && @stat_tracker.find_game_teams("home").count
+    # is it better for this line to be split up into two separate assertions
+    # or for it to be on one line but past the 80char limit?
```
Comment (gabichuelas) [inline test/stat_tracker_test.rb:200]: @rrabinovitch Up to you - i don't know that we'll get dinged for this, it's a small detail.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -117,143 +116,77 @@ def find_team_by_id(id)
...
+    team_avgs = team_averages(sorted_away_team_scores)
+    find_team_by_id(team_with_highest_average_score(team_avgs)).team_name
+  end
 
-    find_team_by_id(highest_scoring_home_id).team_name
+  def highest_scoring_home_team
+    home_teams = find_game_teams("home")
+    sorted_home_team_scores = sort_scores_by_team(home_teams)
+    team_avgs = team_averages(sorted_home_team_scores)
+    find_team_by_id(team_with_highest_average_score(team_avgs)).team_name
```
Comment (gabichuelas) [inline lib/stat_tracker.rb:172]: @rrabinovitch another weird indenting thing here

**Review by gabichuelas — APPROVED** [review]

**Review by araspaas — APPROVED** [review]
looks good to me.

**Comment by sciencefixion** [comment]
Me too!

**Pull request merged**